pkg/kubernetes-mcp-server/cmd/root.go

```diff
@@ -235,10 +245,21 @@ func (m *MCPServerOptions) Run() error {
 		_, _ = fmt.Fprintf(m.Out, "%s\n", version.Version)
 		return nil
 	}
+
+	var oidcProvider *oidc.Provider
+	if m.StaticConfig.AuthorizationURL != "" {
+		provider, err := oidc.NewProvider(context.TODO(), m.StaticConfig.AuthorizationURL)
+		if err != nil {
+			return fmt.Errorf("unable to setup OIDC provider: %w", err)
+		}
+		oidcProvider = provider
+	}
```

Comment:
It seems that oidc.Provider is basically passed along to the mcp.Server instance and in turn is then passed on to the http AuthorizationMiddleware function.

Eventually there's also a call to VerifyToken which in turn performs a call to the Kubernetes VerifyToken function.

There seems to be a lot of coupling here that **might** be possible to extract.

Since the http package is providing auth, IMO mcp should have nothing else to do in this area.

Anyway, this should be done after merging this PR and only if it's possible.

Reply:
With respect to VerifyToken, I'm open to suggestions.

Reply:
> What about this https://github.com/manusa/kubernetes-mcp-server/pull/179?

Not sure yet, but it's a good start.

I'll be creating some more tests for further clarification of the auth flows.
I'll try to provide an alternate proposal (or incremental to #179) once I have a clearer picture.